packages/@aws-cdk/aws-ecs/test/ec2/integ.event-task.lit.expected.json

```diff
@@ -692,13 +692,13 @@
                       "RepositoryName"
                     ]
                   },
-                  ":",
+                  "@sha256:",
```

Comment:
The mix of `Fn::Select` and `Fn::Join` in the `Image` property looks weird to me (+ it ends up with `amazonaws.com` in the final string and not `AWS::URLSuffix`), shouldn't this result in something like:
```json
{
  "Image": {
    "Fn::Join": [
       "",
       [
         { "Ref": "AWS::AccountId" },
         ".dkr.ecr.",
         { "Ref": "AWS::Region" },
         ".",
         { "Ref": "AWS::URLSuffix" },
         "/",
         { "Fn::GetAtt": [ "EventImageAdoptRepositoryDFAAC242", "RepositoryName"] },
         "@sha256:"
         { "Ref": "EventImageImageNameE972A8B1" },
       ] 
    ]
  }
}
```

Reply:
+after pushing, the repo digest includes the full "resolved URI" (with region, accountid, etc.), is it still needed to uses `Ref`s in this case?

Reply:
We're passing the repository name, not full URI, in a parameter. This is because the CustomResource that "adopts" the CDK-toolkit-created ECR repository works on the name (for that is the ECR API). I'm torn myself on whether I should cut the chase and just pass the URI around, but it makes parsing the name out of the URI an awkward process (I'd rather recompose the URI than parse it out).

Reply:
We're passing the repository name, not full URI, in a parameter. This is because the CustomResource that "adopts" the CDK-toolkit-created ECR repository works on the name (for that is the ECR API). I'm torn myself on whether I should cut the chase and just pass the URI around, but it makes parsing the name out of the URI an awkward process (I'd rather recompose the URI than parse it out).